embassy-stm32/src/flash/mod.rs

```diff
@@ -0,0 +1,401 @@
+use crate::pac;
+use crate::peripherals::FLASH;
+use core::convert::TryInto;
+use core::marker::PhantomData;
+use core::ptr::write_volatile;
+use embassy::util::Unborrow;
+use embassy_hal_common::unborrow;
+
+use embedded_storage::nor_flash::{
+    ErrorType, NorFlash, NorFlashError, NorFlashErrorKind, ReadNorFlash,
+};
+
+const FLASH_BASE: usize = 0x8000000;
+
+#[cfg(flash_l4)]
+mod config {
+    use super::*;
+    pub(crate) const FLASH_SIZE: usize = 0x100000;
```

Comment:
Is this the max size for each family? Ideally the size should be the one for the current chip.

Reply:
I've used the 'max size for family' according to the RM, but the true flash size could be lower, and there are a lot of combinations. The true flash size can be found in a system page, which varies with each chip family. Since it cannot be retrieved from the header files directly (without expanding lots of C macros), I wonder if maybe we can create a static map of the page location in stm32-data, and generate a global constant with this address. 

While at it, one could add more of these per-familiy constants in that mapping I guess.

Reply:
Ok, now it's fully derived from XML